### PR TITLE
[WIP] Rewrite the whole thing based on a high-accuracy epoch type

### PR DESCRIPTION
**Pull request by helgee**
I got started with a new accuracy-preserving type. The design is based on the `AbsoluteDate` class from Orekit: https://github.com/CS-SI/Orekit/blob/develop/src/main/java/org/orekit/time/AbsoluteDate.java

**EDIT:** This is basically a straight port of Orekit's time package. *Imitation is the sincerest form of flattery* 😉 

CC: @bgodard, @giordano, @prakharcode 

## ToDo

- [x] String I/O
- [x] Periods & comparisons
- [x] Leap seconds
- [x] Ranges
- [ ] Lots of tests
- [x] Remove old code
- [ ] Docs

### CODE REVIEW

File: test/epochs.jl [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    ep1 = Epoch2{TDB}(ep, 100 * 365.25 * 86400)
+    @test ep.offset == ep1.offset
+
+    ep1 = Epoch2{TDB}(ep, Inf)
+    @test ep1.epoch == typemax(Int64)
+    @test ep1.offset == Inf
+    ep1 = Epoch2{TDB}(ep, -Inf)
+    @test ep1.epoch == typemin(Int64)
+    @test ep1.offset == -Inf
+end
```
Comment (giordano) [inline test/epochs.jl]: New line at the end of the file :slightly_smiling_face:

File: src/Epochs2.jl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        offset′ = (sum - dl) + residual
+        epoch′ = epoch + dl
+    end
+
+    Epoch2{S}(epoch′, offset′)
+end
+
+Epoch2{S}(ep::Epoch2{S}, Δt) where {S} = Epoch2{S}(ep.epoch, ep.offset, Δt)
+
+end
```
Comment (giordano) [inline src/Epochs2.jl]: Missing newline at the end of the file (or extra empty incomplete line) :slightly_smiling_face:

**Comment by helgee** [comment]
@giordano Do you happen to know how one overloads the colon operator?

**Comment by giordano** [comment]
It used to be lowered to the `colon` function in previous versions, now it should be lowered to a function called `:` according to the release notes.

**Comment by helgee** [comment]
...and we have ranges 🎉

**Pull request merged**